apps/api/src/app/workflows-v2/usecases/create-variables-object/create-variables-object.usecase.ts

```diff
@@ -24,7 +39,35 @@ export class CreateVariablesObject {
     ];
     const showIfVariables = this.extractMailyAttribute(controlValues, MailyAttrsEnum.SHOW_IF_KEY);
 
-    return keysToObject(variables, arrayVariables, showIfVariables);
+    const variablesObject = keysToObject(variables, arrayVariables, showIfVariables);
+    if (isEnhancedDigestEnabled) {
+      return this.ensureEventsArray(variablesObject);
+    }
+
+    return variablesObject;
+  }
+
+  private ensureEventsArray(variablesObject: Record<string, unknown>) {
```

Comment:
Whenever the `steps.digest-step.events, steps.digest-step.events.length, steps.digest-step.eventCount` variables are used we need to ensure that the `events` array variable is generated. Because the "count" and "events" variables are "connected" to each other.

Reply:
I feel that this function can have a more descriptive name such as ensureEventsVariableIsAnArray